### PR TITLE
ARROW-7312: [Rust] Implement std::error::Error for ArrowError.

### PR DESCRIPTION
**Pull request by liurenjie1024**
Add this implementation so that others can handle errors from arrow crate better.

### CODE REVIEW

**Comment by liurenjie1024** [comment]
@andygrove @nevi-me @sunchao @paddyhoran Please help to review.

File: rust/arrow/src/error.rs [hunk 1]
```diff
@@ -70,4 +71,12 @@ impl From<::std::string::FromUtf8Error> for ArrowError {
     }
 }
 
+impl Display for ArrowError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Arrow error happened!")
```
Comment (nevi-me) [inline rust/arrow/src/error.rs]: Why not display the type of and details ArrowError?

Comment (liurenjie1024) [reply rust/arrow/src/error.rs]: Details should be displayed in debug mode.

Comment (nevi-me) [reply rust/arrow/src/error.rs]: I still don't understand, but perhaps it's because I'm not familiar with the Display trait.

**Review by nevi-me — APPROVED** [review]

File: rust/arrow/src/error.rs (hunk 1 above)
Comment (paddyhoran) [reply rust/arrow/src/error.rs]: Yea, I'm with @nevi-me here.  His point is that "Arrow error happened!" is not that informative.  You would also want to know the variant of the enum and any payload, right?

Comment (liurenjie1024) [reply rust/arrow/src/error.rs]: fixed.

**Review by paddyhoran — APPROVED** [review]
LGTM, thanks @liurenjie1024.  I don't see CI failing but I guess we should wait before merging.

**Comment by nevi-me** [comment]
AppVeyor no longer runs any Rust tests, maybe we should remove it from the Rust runs (I haven't checked how it gets triggered though, so maybe that's not possible). Is it possible @kou?

**Comment by kou** [comment]
Rust job on AppVeyor has been migrated to GitHub Actions based job. So we don't need to wait for AppVeyor result for changes that only affects to Rust.

FYI: We can disable AppVeyor job check for Rust only changes entirely by the following change:

```diff
diff --git a/appveyor.yml b/appveyor.yml
index a6d79de7e..cd8617893 100644
--- a/appveyor.yml
+++ b/appveyor.yml
@@ -31,7 +31,6 @@ only_commits:
     - python/
     - r/
     - ruby/
-    - rust/
 
 cache:
   - C:\Users\Appveyor\clcache1
```

**Pull request closed without merging**